### PR TITLE
added add to cart operation

### PR DESCRIPTION
**Pull request by nginx500**
Task: https://github.com/learning-project-01/bookstore-app/issues/13

image-1
curl --location --request POST 'localhost:8080/cart/1'

image-2
curl --location 'localhost:8080/cart'

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/controllers/CartController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+
+  @Autowired
+  private CartItemService cartItemService;
+
```
Comment (nginx500) [inline src/main/java/com/example/bookstoreapp/controllers/CartController.java:19]: this is add catalog Item Tocart

File: src/main/java/com/example/bookstoreapp/controllers/CartController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  @PostMapping("/{catalogItemId}")
+  public CartItem addItemToCart(@PathVariable("catalogItemId") Long catalogItemId){
+    return cartItemService.addItem(catalogItemId);
+  }
+
```
Comment (nginx500) [inline src/main/java/com/example/bookstoreapp/controllers/CartController.java:24]: get cart details

File: src/main/java/com/example/bookstoreapp/entities/CartItemEntity.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+
+@Entity
+@Table(name = "cart_item")
+@Data
+public class CartItemEntity {
```
Comment (nginx500) [inline src/main/java/com/example/bookstoreapp/entities/CartItemEntity.java:11]: This is the mapping class to store the CartItem

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CartItemServiceImpl implements CartItemService {
```
Comment (nginx500) [inline src/main/java/com/example/bookstoreapp/services/impl/CartItemServiceImpl.java:25]: This is service class implementation

File: src/main/java/com/example/bookstoreapp/utils/IdGenerator.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.example.bookstoreapp.utils;
+
+import java.util.UUID;
+
```
Comment (nginx500) [inline src/main/java/com/example/bookstoreapp/utils/IdGenerator.java:4]: Used to give the unique id , but this is not a good approach in a distributed system as it is based on timestamp.

https://chronicle.software/unique-timestamp-identifiers/

**Comment by Subhransupanda2000** [comment]
why did you add cartItemstate. wht r  uses of it

**Comment by rkpatra201** [comment]
> why did you add cartItemstate. wht r  uses of it

1. Added an item called ABC into cart.
2. I want to buy ABC later. So I can do save for later.
3. As it is in save later, when I want to buy I will do in cart from save later for item ABC. 
4. And the buy now state is here: let say I added DEF and GHK to cart but want to purchase only DEF. In this case buy now state will set for DEF and in cart state for GHK.
5. For more details see amazon cart.

**Pull request merged**